reconcile/gitlab_housekeeping.py

```diff
@@ -18,10 +20,35 @@
                'do-not-merge/hold', 'do-not-merge/pending-review']
 
 QONTRACT_INTEGRATION = 'gitlab-housekeeping'
+DATE_FORMAT = '%Y-%m-%dT%H:%M:%S.%fZ'
+
+
+def clean_pipelines(dry_run, gl_project, pipelines, pipeline_timeout=60):
+    now = time.time()
+
+    pending_pipelines = [p for p in pipelines
+                         if p['status'] in ['pending', 'running']]
+
+    if not pending_pipelines:
+        return
+
+    for p in pending_pipelines:
+        update_time = time.strptime(p['updated_at'], DATE_FORMAT)
+
+        elapsed = now - time.mktime(update_time)
+
+        # pipeline_timeout converted in seconds
+        if elapsed > pipeline_timeout * 60:
+            logging.info(['canceling', p['web_url']])
+            if not dry_run:
+                try:
+                    gl_project.pipelines.get(p['id']).cancel()
+                except gitlab.exceptions.GitlabPipelineCancelError as err:
```

Comment:
I'm not sure I'd catch the exception here. This hides any failures from the caller, and prevents them from taking any corrective action. Why not let it pass?